Update teams model

rotem2022 opened the pull request:
update teams entity.
The purpose of this entity is to divide the participants in an event into teams (such as in the case of basketball where two teams are required).

changing files:
- models.py
- tests.py

created teams entity according to the ERD.

resolve #25

talko611 commented on teams/admin.py:
```diff
@@ -0,0 +1,3 @@
+# from django.contrib import admin
+
+# Register your models here.
```
You need to register your model to the admin site

talko611 replied:
You need to register the app to the admin app

ygalblum reviewed (commented):
Please see and fix the flake8 errors

rotem2022 commented on teams/admin.py:
```diff
@@ -0,0 +1,4 @@
+from django.contrib import admin
+from .models import Teams
+
+admin.site.register(Teams)
```
yes that is right, thank you

kobihk commented:
@rotem2022 please fix the typo in the PR description:
tests.py

kobihk commented:
@rotem2022 please rebase and resolve the conflicts

kobihk commented:
> * tsest.py

kobihk commented:
@rotem2022 please fix the typo in the PR description:
tsest.py

ormergi commented on teams/models.py:
```diff
@@ -0,0 +1,8 @@
+class Teams(models.Model):
+    name = models.CharField(max_length=30)
+
+    def __str__(self) -> str:
```
Why do we need this method?

If it is not used anywhere, you can remove it for now and add it when needed.

rotem2022 replied:
Ok, I removed this function.

ormergi reviewed (changes requested):
Please fix the commit message title and format, the describption should be in the middle and  the sign-off at the bottom.
Also please change the description to elaborate what this change does and why you need it.

rotem2022 commented:
> Please fix the commit message title and format, the describption should be in the middle and the sign-off at the bottom. Also please change the description to elaborate what this change does and why you need it.

I changed the commit description and added some explanation.

ormergi commented:
> I changed the commit description and added some explanation.

I see that you added an explanation to the PR description but the commit message is empty.
Please add an explanation to the commit message as well.
Also, you can use the `-s` flag to add the standard sign-off instead of writing it manually: 
`git commit -s` / `git commit --amend -s.

rotem2022 commented:
> > I changed the commit description and added some explanation.
> 
> I see that you added an explanation to the PR description but the commit message is empty. Please add an explanation to the commit message as well. Also, you can use the `-s` flag to add the standard sign-off instead of writing it manually: `git commit -s` / `git commit --amend -s.

All done.

ygalblum commented on fit_meet/settings.py:
```diff
@@ -43,6 +40,7 @@
     'category.apps.CategoryConfig',
     'category_location.apps.CategoryLocationConfig',
     'poll.apps.PollConfig',
+    'teams.apps.TeamsConfig'
```
Commas are allowed also on the last element. So, it's safer to add a comma to avoid creating issues to other PRs
```suggestion
    'teams.apps.TeamsConfig',
```

rotem2022 replied:
Yes I understand. I made the change.

kobihk commented on teams/tests.py:
```diff
@@ -0,0 +1,21 @@
+
+
+@pytest.fixture
+def create_team():
```
in this fixture, you create a specific team object 
so I think it will be better to call it like:
```suggestion
def team1():
```

kobihk commented on teams/tests.py:
```diff
@@ -0,0 +1,21 @@
+
+@pytest.fixture
+def create_team():
+    team = Teams.objects.create(name='Test Team')
+    return team
```
I would define the name as a constant like:
TEAM1 = 'Team 1' 
because if you will want to check it, it will be easier to use the constant
```suggestion
    return Teams.objects.create(name=TEAM1)
```

kobihk commented on teams/tests.py:
```diff
@@ -0,0 +1,21 @@
+@pytest.mark.django_db
+class TestTeamsModel:
+
+    def test_create_team(self, create_team):
+        create_team.save()
+        assert create_team in Teams.objects.all()
```
here it will be more readable like:
```suggestion
    def test_create_team(self, team1):
        team1.save()
        assert team1 in Teams.objects.all()
```

kobihk commented on teams/tests.py:
```diff
@@ -0,0 +1,21 @@
+        create_team.save()
+        assert create_team in Teams.objects.all()
+
+    def test_delete_team(self, create_team):
+        create_team.save()
+        create_team.delete()
+        assert create_team not in Teams.objects.all()
```
same ^^

kobihk commented on teams/tests.py:
```diff
@@ -0,0 +1,21 @@
+
+@pytest.mark.django_db
+class TestTeamsModel:
+
```
I think you can make more tests like for validation of team name:
 - long name
 - empty name
 - a name with special characters
 - same name
for most of the tests, you can use pytest.mark.parametrize as you can see for example in:
https://github.com/EdDev/django-testing-tutorial/blob/main/member/tests.py#L146

rotem2022 replied:
what do you mean by same name?

kobihk replied:
Can I create a team name with the exact same name? 
like 2 teams with the same name "Team 1"

ormergi commented on teams/tests.py:
```diff
@@ -0,0 +1,21 @@
+@pytest.mark.django_db
+class TestTeamsModel:
+
+    def test_create_team(self, create_team):
+        create_team.save()
+        assert create_team in Teams.objects.all()
+
+    def test_delete_team(self, create_team):
+        create_team.save()
+        create_team.delete()
+        assert create_team not in Teams.objects.all()
```
Please note that these tests don't test the app business login, it tests Django built-in ORM functionality.
I would remove them because they don't test the app business logic.

rotem2022 replied:
Ok, I removed them.

kobihk commented:
@rotem2022 AFAICS the title and description of this PR should be updated because you already created the event entity in #52

kobihk commented on teams/tests.py:
```diff
@@ -1,21 +1,19 @@
+        with pytest.raises(ValidationError):
+            Teams.objects.create(name=LONG_NAME).full_clean()
 
-    def test_delete_team(self, create_team):
-        create_team.save()
-        create_team.delete()
-        assert create_team not in Teams.objects.all()
+    def test_create_empty_name_team(self):
+        with pytest.raises(ValidationError):
+            Teams.objects.create(name=EMPTY_NAME).full_clean()
```
as you can see the code in both tests looks the same except for the input of the name 
for this reason, you have the option to use pytest.mark.parametrize 
as you can see for example in:
https://github.com/EdDev/django-testing-tutorial/blob/main/member/tests.py#L146

kobihk replied:
you can also add a method that checks the name with special characters

rotem2022 replied:
done.

rotem2022 commented:
> @rotem2022 AFAICS the title and description of this PR should be updated because you already created the event entity in #52

All done.

Sendy459 reviewed (approved):
LGTM

rotem2022 commented on teams/models.py:
```diff
@@ -1,5 +1,13 @@
 class Teams(models.Model):
     name = models.CharField(max_length=30)
+
+    def save(self, *args, **kwargs):
```
I added a test for the save.
